flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/CorrelateCodeGenerator.scala

```diff
@@ -65,9 +68,11 @@ object CorrelateCodeGenerator {
 
     // according to the SQL standard, every scalar function should also be a table function
     // but we don't allow that for now
-    if (!rexCall.getOperator.isInstanceOf[BridgingSqlFunction] &&
-        !rexCall.getOperator.isInstanceOf[TableSqlFunction]) {
-      throw new ValidationException("Currently, only table functions can emit rows.")
+    rexCall.getOperator match {
+      case func: BridgingSqlFunction if func.getDefinition.getKind == FunctionKind.TABLE => //ok
```

Comment:
nit: whitespace
could be moved into a hotfix commit and merged immediately

Reply:
I'd prefer to merge it as part of the PR. I added a test for that case which I would have problems extracting. If you think it is worth doing it, I will try to.